ouroboros-network-api/src/Ouroboros/Network/PeerSelection/LocalRootPeers.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+module Ouroboros.Network.PeerSelection.LocalRootPeers (OutboundConnectionsState (..)) where
+
+import GHC.Generics
+import NoThunks.Class
+
+data OutboundConnectionsState =
+    TrustedStateWithExternalPeers
+    -- ^
+    -- * /in bootstrap mode/: connected only to trusted local peers and at least
+    --   one bootstrap peer.
+    -- * /in genesis mode/: meeting target of active big ledger peers, ledger
+    --   state judgement must be `TooOld`.
+
+  | EnteringTrustedStateWithExternalPeers
+    -- ^ entering trusted state with eternal peers.
+    --
+    -- * /in bootstrap mode/: `requiredBootstrapPeers` returns `True` and has
+    --   at least on bootstrap peer.
+    -- * /in genesis mode/: ledger state judgement returns `TooOld`
+
+  | HiddenRelayOrBP
+    -- ^ A node is classified as a hidden relay or a BP if it is configured so
+    -- that it can only have a chance to be connected to local roots. This is
+    -- true in one of the two ways:
+    --
+    -- * it's configured with `DontUseBootstrapPeers` and `DontUseLedgerPeers`; or
+    -- * the targets are set so the node can only be connected to local roots.
+
+  | UntrustedState
+    -- ^ catch all other cases
+  deriving (Eq, Show, Generic)
```

Comment:
We also need `TrustedUnreachableState` for peers not configured with any trusted peers (bootstrap or local roots).

Reply:
We decided to map the promiscuous mode to `TrustedStateWithExternalPeers` to simplify things for `consensus`.

Reply:
FWIW if one has a weird configuration like specifying node A as a local root and also adding node A to `UseBootstrapPeers` won't be a problem since we prioritize local roots so the node will respect the invariant that the bootstrap peers and local root peers should be disjoint sets